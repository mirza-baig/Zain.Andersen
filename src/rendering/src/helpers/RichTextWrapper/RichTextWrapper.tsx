// Global
import { RichTextProps } from '@sitecore-jss/sitecore-jss-react';
import { RichText } from '@sitecore-jss/sitecore-jss-nextjs';
// Lib
import useExperienceEditor from 'lib/utils/use-experience-editor';
import classNames from 'classnames';

type RichTextWrapperProps = RichTextProps & {
  classes?: string;
  refer?: string;
};
const RichTextWrapper = ({
  field,
  classes,
  refer = 'body-copy',
  ...props
}: RichTextWrapperProps): JSX.Element => {
  const isEE = useExperienceEditor();

  // Just pass as normal if in Experience Editor
  if (isEE) {
    return (
      <div className={classNames(refer, classes)}>
        <RichText field={field} {...props} />
      </div>
    );
  }

  // Bail if we don't have any field data
  if (!field || !field.value) {
    return <></>;
  }

  return (
    <div className={classNames(refer, classes)}>
      <RichText field={{ value: field.value }} {...props} />
    </div>
  );
};

export default RichTextWrapper;
