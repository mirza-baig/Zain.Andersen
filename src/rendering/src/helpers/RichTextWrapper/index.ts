export { default as RichTextWrapper } from './RichTextWrapper';
export type { RichTextProps } from '@sitecore-jss/sitecore-jss-react';
