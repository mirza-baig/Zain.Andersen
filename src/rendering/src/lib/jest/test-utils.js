// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
