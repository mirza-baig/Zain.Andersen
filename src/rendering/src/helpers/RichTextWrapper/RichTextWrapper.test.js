// Lib
import { snapshot, renderComponent, eeLayoutData } from 'lib/jest/test-utils';
// Local
import RichTextWrapper from './RichTextWrapper';
import defaultData, { regExTest } from './RichTextWrapper.mock-data';

const NEW_TAB_TEXT = '(Opens in a new tab)';

it('renders common usage correctly', () => {
  const component = snapshot(RichTextWrapper, { componentProps: defaultData });
  // Expect that we have inserted our new tab content
  const newTabText = component.queryAllByText(NEW_TAB_TEXT);
  expect(newTabText).not.toBe(null);
});

it('handles various attributes, invalid attributes, and character types in <a> tags appropriately and nested tags in <a> tags', () => {
  snapshot(RichTextWrapper, { componentProps: regExTest });
});

it("doesn't change content in Experience Editor", () => {
  // const component = renderComponent(RichTextWrapper, {
  //   componentProps: defaultData,
  //   layoutData: eeLayoutData,
  // });
  // // Expect that our new tab SR text is not inserted
  // const newTabText = component.queryByText(NEW_TAB_TEXT);
  // expect(newTabText).toBe(null);
});

it("doesn't display with no field data", () => {
  const component = renderComponent(RichTextWrapper, {});
  expect(component.container.firstChild).toBe(null);
});
