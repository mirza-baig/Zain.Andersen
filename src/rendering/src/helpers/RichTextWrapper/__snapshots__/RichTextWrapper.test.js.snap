// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles various attributes, invalid attributes, and character types in <a> tags appropriately and nested tags in <a> tags 1`] = `
<div
  class="body-copy"
>
  <div>
    <p>
      At vero eos et accusamus et iusto odio dignissimos ducimus qui 
      <a
        class="overlay-source"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        blanditiis praesentium
      </a>
       voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.
    </p>
     
    <ul>
       
      <li>
        Et harum quidem rerum facilis 
        <a
          href="https://www.google.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          est et expedita
        </a>
         distinctio.
      </li>
       
      <li>
        Nam libero tempore, cum soluta nobis est eligendi optio cumque.
      </li>
       
    </ul>
     
    <p>
      Nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et

    </p>
    <p>
      voluptatum

      <a
        href="adfdf"
        rel="adfd"
        target="_blank"
      >
        molestiae non recusandae
      </a>
      .
    </p>
    

    <a
      bb="dafa"
      crazy="crazy"
      target="_blank"
    >
      foobar 
      <span>
        foobar
      </span>
      &lt;&gt;&lt;&gt;&lt;&gt;&lt;&gt;&gt;&gt;&gt;&gt;&gt;&lt;&lt;&lt;#*)#(*Q$)JALKJFLKJALKSFJ
    </a>
    

    <a
      bb="lakdfjl"
      target="_blank"
      zz="adlkfjalaldkfj"
    />
    

    <a
      bb="lakdfjl"
      data-foo="adlkfjalaldkfj"
      target="_blank"
    >
      adfadfadf 
      <span>
        afadfd
      </span>
    </a>
    

    <a
      dababd="zzzz"
      target="_blank"
    />
    

    <a
      dababd="zzzz"
    />
    

    <a
      11111="zzzz"
      target="_blank"
    />
    

    <a
      #%%#)(*)(%*="zzz"
      target="_blank"
    >
      <div>
        foobarbaz
      </div>
    </a>
    

    <a
      #%%#)(*)(%*="zzz"
      target="_blank"
    >
      

      <div>
        foobarbaz
      </div>
      

    </a>
    <a
      #%%#)(*)(%*="zzz"
    >
      

      <div>
        foobarbaz
      </div>
      

    </a>
    <a
      #%%#)(*)(%*="zzz"
      target="_blank"
    >
      

      <div>
        foobarbaz
      </div>
      

    </a>
    
Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.
    <p />
    -
  </div>
</div>
`;

exports[`renders common usage correctly 1`] = `
<div
  class="body-copy"
>
  <div>
    <p>
      At vero eos et accusamus et iusto odio dignissimos ducimus qui 
      <a
        class="overlay-source"
        href="https://www.google.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        blanditiis praesentium
      </a>
       voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga.
    </p>
     
    <ul>
       
      <li>
        Et harum quidem rerum facilis 
        <a
          href="https://www.google.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          est et expedita
        </a>
         distinctio.
      </li>
       
      <li>
        Nam libero tempore, cum soluta nobis est eligendi optio cumque.
      </li>
       
    </ul>
     
    <p>
      Nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et 
      <a
        href="https://www.google.com"
      >
        molestiae non recusandae
      </a>
      . Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.
    </p>
  </div>
</div>
`;
